inject-generator/src/main/java/io/avaje/inject/generator/Util.java

```diff
@@ -247,4 +250,134 @@ public static String trimMethod(String method) {
       .replace('.', '_')
       .replace(Constants.DI, "");
   }
+
+  public static void buildBeanConditional(
+      Append writer,
+      Set<String> conditionTypes,
+      Set<String> missingTypes,
+      Set<String> qualifierNames) {
+    if (!conditionTypes.isEmpty()
+        || !qualifierNames.isEmpty()
+        || !missingTypes.isEmpty()) {
+
+      writer.append("    if (");
+      var first = true;
+      for (final var conditionType : conditionTypes) {
+
+        if (first) {
+
+          writer.append("!builder.contains(%s.class)", Util.shortName(conditionType));
+          first = false;
+          continue;
+        }
+
+        writer.append(" || !builder.contains(%s.class)", Util.shortName(conditionType));
+      }
+
+      for (final var missing : missingTypes) {
+
+        if (first) {
+
+          writer.append("builder.contains(%s.class)", Util.shortName(missing));
+          first = false;
+          continue;
+        }
+
+        writer.append(" || builder.contains(%s.class)", Util.shortName(missing));
+      }
+
+      for (final var qualifier : qualifierNames) {
+
+        if (first) {
+
+          writer.append("!builder.containsQualifier(\"%s\")", qualifier);
+          first = false;
+          continue;
+        }
+
+        writer.append(" || !builder.containsQualifier(\"%s\")", qualifier);
+      }
+
+      writer.append(")").eol().append("     return;").eol().eol();
+    }
+  }
+
+  public static void buildPropertyConditional(
+      Append writer,
+      Set<String> containsProps,
+      Set<String> missingProps,
+      Map<String, String> propertyEquals,
+      Map<String, String> propertyNotEquals) {
+
+    final var propertyGetter = "Config.getOptional(\"%s\")";
```

Comment:
I'm pretty sure we want to add 1 more level of abstraction.  A new interface (RequiresPropertyPlugin ?) that has methods:

- contains(String)
- equalTo(String, String)
- notEqualTo(String, String)


And then implement that RequiresPropertyPlugin using avaje Config and fallback to System properties.

Reply:
Yeah I like this, will add to avaje config